lib/gps.js

```diff
@@ -0,0 +1,439 @@
+//var IS_TEST_MODE = !!process.env.IS_TEST_MODE;
+var Board = require("./board");
+var events = require("events");
+var util = require("util");
+var Pin = require("./pin");
+
+var priv = new Map();
+
+var Breakouts = {
+
+  /*
+   * https://www.adafruit.com/products/746
+   */
+  ADAFRUIT_ULTIMATE_GPS: {
+    receiver: {
+      value: "FGPMMOPA6H"
+    }
+  }
+
+};
+
+// GPS Antenna Modules
+var Receivers = {
+
+  /*
+   * http://www.gtop-tech.com/en/product/LadyBird-1-PA6H/MT3339_GPS_Module_04.html
+   */
+  FGPMMOPA6H: {
+    // Later, when we add logging that code will go here
+    chip: {
+      value: "MT3339"
+    }
+  }
+
+};
+
+// GPS chips
+var Chips = {
+
+  DEFAULT: {
+    baud: {
+      value: 9600
+    },
+    configure: {
+      value: function(callback) {
+        callback();
+      }
+    }
+  },
+
+  /*
+   * http://www.mediatek.com/en/products/connectivity/gps/mt3339/
+   */
+  MT3339: {
+    baud: {
+      value: 9600
+    },
+    configure: {
+      value: function(callback) {
+        callback();
+      }
+    },
+    frequency: {
+      get: function() {
+        var state = priv.get(this);
+        return state.frequency;
+      },
+      set: function(freq) {
+        var state = priv.get(this);
+
+        // Enforce maximum frequency of 10hz
+        if (freq < 100) {
+          freq = 100;
+        }
+
+        this.sendCommand("$PMTK220,"+String(freq));
+        state.frequency = freq;
+      }
+    },
+    restart: {
+      // Reboot the receiver
+      value: function(coldRestart) {
+
+        if (coldRestart === true) {
+          this.sendCommand("$PMTK103");
+        } else {
+          this.sendCommand("$PMTK101");
+          setTimeout(function() {
+            this.sendCommand("");
+          }.bind(this), 1000);
+        }
+
+      }
+    }
+  }
+
+};
+
+
+/**
+ *
+ * @constructor
+ *
+ * @param {Object} opts Options: pin(s), chip, receiver, breakout, precision, serialport, freq
+ *
+ * Sample initialization
+ *
+ *    new five.GPS({ pins: {rx: 10, tx: 11});
+ *
+ */
+
+function GPS(opts) {
+
+  var breakout, receiver, chip, state;
+
+  if (!(this instanceof GPS)) {
+    return new GPS(opts);
+  }
+
+  // Allow users to pass in a 2 element array for tx and rx pins
+  if (Array.isArray(opts)) {
+    opts = { pins: {tx: opts[0], rx: opts[1], onOff: opts[2] }};
```

Comment:
Boards always print these in the order: RX, TX. 

Also: 

```c
SoftwareSerial mySerial(10, 11); // RX, TX
```

Therefore, `GPS` should expect them in this order: 

```js
five.GPS([ RX, TX, onOff ]);
```